internal/shellgen/scripts.go

```diff
@@ -110,6 +110,11 @@ func writeInitHookFile(devbox devboxer, body, tmpl, filename string) (err error)
 	}
 	defer script.Close() // best effort: close file
 
+	if body == devconfig.DefaultInitHook || strings.TrimSpace(body) == "" {
```

Comment:
Mind adding a comment to the effect of:
```
// skip adding init-hook recursion guard for the 
// default hook or any empty hook
```

As is, not obvious why the template below is being skipped.

Will also be a good idea to comment somewhere in the template that it will be skipped for default or empty hooks, in case we later introduce some other logic for init-hooks in the template files.